src/local/butler/reproduce.py

```diff
@@ -208,6 +209,28 @@ def _get_testcase_id_from_url(testcase_url):
   return testcase_id
 
 
+def _prepare_environment_for_android(testcase_path):
+  """Additional environment overrides needed to run on an Android device."""
+  environment.set_value('OS_OVERRIDE', 'ANDROID')
+
+  # Bail out if we don't have an Android device connected.
+  if not environment.get_value('ANDROID_SERIAL'):
+    # TODO(mbarbella): Handle the one-device case gracefully.
+    raise errors.ReproduceToolUnrecoverableError('Please set ANDROID_SERIAL.')
+
+  # TODO(mbarbella): Should we run device.initialize_device?
+
+  # Push the test case and build APK to the device.
+  apk_path = environment.get_value('APP_PATH')
+  device.install_application_if_needed(apk_path, True)
+
+  testcase_manager.get_command_line_for_application(
+      write_command_line_file=True)
+
```

Comment:
PKG_NAME is no set, which will cause gestures to not work. 

https://github.com/google/clusterfuzz/blob/master/src/python/platforms/android/gestures.py#L39